### PR TITLE
add uniqThetaSketch

### PR DESCRIPTION
**Pull request by pingyu**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature  #14893 

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
- Add uniqThetaSketch to support [Theta Sketch](https://datasketches.apache.org/docs/Theta/ThetaSketchFramework.html) in ClickHouse.

Detailed description / Documentation draft:
- Added document at docs/en/sql-reference/aggregate-functions/reference/uniqthetasketch.md

### CODE REVIEW

**Comment by pingyu** [comment]
The failure of build seems to be caused by DataSketches submodule is not checked out.
How can I fix it ? @Avogar

**Comment by Avogar** [comment]
Add the needed submodule in this list: https://github.com/ClickHouse/ClickHouse/blob/30f2d839c34208db69132e063ba669aacebbcd2c/docker/test/fasttest/run.sh#L187

File: src/AggregateFunctions/ThetaSketchData.h [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+
+template <typename Key>
+class ThetaSketchData : private boost::noncopyable
+{
+private:
+    std::unique_ptr<datasketches::update_theta_sketch> sk_update;
+    std::unique_ptr<datasketches::theta_union> sk_union;
+
+    inline datasketches::update_theta_sketch * get_sk_update()
```
Comment (Avogar) [inline src/AggregateFunctions/ThetaSketchData.h:20]: We use camelCase for function and method names.

**Comment by Avogar** [comment]
You also should support case when ENABLE_DATASKETCHES=0 in code. Now it just cannot build with it. Put all logic with submodule under `if USE_DATASKETCHES` and throw an exception when it's not enabled and uniqThetaSketch is called.

**Comment by pingyu** [comment]
Comments addressed. PTAL, thanks~ @Avogar

File: src/AggregateFunctions/AggregateFunctionUniq.h [hunk 2]
```diff
@@ -124,6 +125,26 @@ struct AggregateFunctionUniqExactData<String>
...
+struct AggregateFunctionUniqThetaSketchData
+{
+    using Set = ThetaSketchData<UInt64>;
+    Set set;
+
+    static String getName() { return "uniqThetaSketch"; }
+};
+
+/// For a function that takes multiple arguments. Such a function pre-hashes them in advance, so TrivialHash is used here.
+struct AggregateFunctionUniqThetaSketchDataForVariadic
```
Comment (Avogar) [inline src/AggregateFunctions/AggregateFunctionUniq.h:139]: This struct is completely the same as `AggregateFunctionUniqThetaSketchData`, why do we need it?

File: cmake/find/datasketches.cmake [hunk 3]
```diff
@@ -22,6 +22,8 @@ endif()
 
 if (DATASKETCHES_LIBRARY AND DATASKETCHES_INCLUDE_DIR)
     set(USE_DATASKETCHES 1)
+else()
+    set(USE_DATASKETCHES 0)
```
Comment (Avogar) [inline cmake/find/datasketches.cmake:26]: It seems that in's not necessary to write this else. Or I'm wrong?

File: src/AggregateFunctions/AggregateFunctionUniq.h (hunk 2 above)
Comment (pingyu) [reply src/AggregateFunctions/AggregateFunctionUniq.h]: Ah...we do not need it. It's my carelessness. Comment addressed, PTAL~

File: cmake/find/datasketches.cmake (hunk 3 above)
Comment (pingyu) [reply cmake/find/datasketches.cmake]: Yes, you are write. I removed the `else` statements, PTAL~

**Comment by Avogar** [comment]
LGTM, some problems with synchronization again, I will fix it and will merge.

**Comment by Avogar** [comment]
I put all tests with uniqThetaSketch in one file to have an ability to disable it when we cannot use datasketches submodule. Lets wait for the tests and if there are no errors - merge.

**Pull request merged**